clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -2072,6 +2073,9 @@ void CIRGenModule::emitTopLevelDecl(Decl *decl) {
         emitTopLevelDecl(childDecl);
     break;
   }
+  case Decl::FileScopeAsm:
+    buildFileScopeAsm(cast<FileScopeAsmDecl>(decl));
```

Comment:
Just call `moduleAsm.push_back(Asm.str());` directly here, the two levels of indirection are not providing any benefits right now